### PR TITLE
Add skuba auth login

### PR DESCRIPTION
**Pull request by jhsiaosuse**
Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

**Comment by jhsiaosuse** [comment]
@innobead @c3y1huang @cclhsu @nkoranova review it plz.

File: adoc/admin-security-rbac.adoc [hunk 1]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+. Gangway returns a link to download `kubeconfig` or self-configure `kubectl` instructions to the user.
 +
 image::rbac-configure-kubectl.png[]
 
 
-. User self-configures `kubectl`.
-. The `kubectl` connects to {kube} API-server.
-. {kube} CRDs validate {kube} API-server request and return response.
-. The `kubectl` connects to authorized {kube} resources through {kube} API-server.
+. User download `kubeconf` or self-configure `kubectl`.
```
Comment (innobead) [inline adoc/admin-security-rbac.adoc]: downloads

File: adoc/admin-security-rbac.adoc [hunk 2]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+. User use `kubectl` to connect {kube} apiserver.
```
Comment (innobead) [inline adoc/admin-security-rbac.adoc]: uses

File: adoc/admin-security-rbac.adoc [hunk 3]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+. The `kubectl` connects to authorized {kube} resources through {kube} apiserver.
+
+=== CLI Flow
+image::oidc_flow_cli.png[]
+// Source: suse-rbac-oidc-flow-cli.xml (open with http://draw.io/app)
+
+. User requests access through `skuba auth login` with Dex server URL, login username and password.
+. Dex receives login username and password, and do login and approve access request to the connected identity providers(connectors).
+. Dex continues with OIDC authentication flow on behalf of the user and creates/updates data to {kube} CRDs.
+. Dex response the ID token and refresh token to `skuba auth login`.
```
Comment (innobead) [inline adoc/admin-security-rbac.adoc]: `response` -> `responds`

File: adoc/admin-security-rbac.adoc [hunk 4]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+. `skuba auth login` generate the kubeconfig file `kubeconf.txt`.
```
Comment (innobead) [inline adoc/admin-security-rbac.adoc]: `generate` -> `generates`

File: adoc/admin-security-rbac.adoc [hunk 5]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+. User use `kubectl` to connect {kube} apiserver.
```
Comment (innobead) [inline adoc/admin-security-rbac.adoc]: `use` -> `uses`

File: adoc/admin-security-rbac.adoc [hunk 6]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
 == Authentication Flow
 
-Authentication is composed of two open source libraries:
+Authentication is composed of:
 
 * link:https://github.com/dexidp/dex[Dex] is an identity provider service (idP) that uses link:https://openid.net/connect/[OIDC] (Open ID Connect) to drive authentication for client application. It acts as portal to defer authentication to provider through connected identity providers(connectors).
-* link:https://github.com/heptiolabs/gangway[Gangway], a client application that enables authentication flow for your {productname}. User can login, authorize access and self-configure `kubectl`.
+* Client
+  . Web browser: link:https://github.com/heptiolabs/gangway[gangway], a web application that enables authentication flow for your {productname}. User can login, authorize access, download `kubeconfig` or self-configure `kubectl`.
+  . Command line: `skuba auth login`, a CLI application that enables authentication flow for your {productname}. User can login, authorize access, and got `kubeconfig`.
```
Comment (cclhsu) [inline adoc/admin-security-rbac.adoc]: change `got` to `get`

File: adoc/admin-security-rbac.adoc [hunk 7]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+// Source: suse-rbac-oidc-flow-web.xml (open with http://draw.io/app)
 
 . User requests access through Gangway.
 . Gangway redirects to Dex.
 . Dex redirects to connected identity provider (connector). User login and approve access request are generated.
-. Dex continues with OIDC authentication flow on behalf of the user and creates/updates data to {kube}  CRDs.
+. Dex continues with OIDC authentication flow on behalf of the user and creates/updates data to {kube} CRDs.
 . Dex redirects the user to Gangway. This redirect includes (ID/refresh) tokens.
-. Gangway returns `kubectl` self-configure instructions to the user.
+. Gangway returns a link to download `kubeconfig` or self-configure `kubectl` instructions to the user.
```
Comment (cclhsu) [inline adoc/admin-security-rbac.adoc]: change `self-configure` to `self-configuring`

File: adoc/admin-security-rbac.adoc [hunk 8]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
 +
 image::rbac-configure-kubectl.png[]
 
 
-. User self-configures `kubectl`.
-. The `kubectl` connects to {kube} API-server.
-. {kube} CRDs validate {kube} API-server request and return response.
-. The `kubectl` connects to authorized {kube} resources through {kube} API-server.
+. User downloads `kubeconf` or self-configure `kubectl`.
```
Comment (cclhsu) [inline adoc/admin-security-rbac.adoc]: User downloads `kubeconfig` or self-configures with `kubectl`.

File: adoc/admin-security-rbac.adoc [hunk 9]
```diff
@@ -6,34 +6,50 @@ RBAC uses the `rbac.authorization.k8s.io` API group to drive authorization decis
...
+. User uses `kubectl` to connect {kube} apiserver.
+. {kube} CRDs validate {kube} apiserver request and return response.
+. The `kubectl` connects to authorized {kube} resources through {kube} apiserver.
+
+=== CLI Flow
+image::oidc_flow_cli.png[]
+// Source: suse-rbac-oidc-flow-cli.xml (open with http://draw.io/app)
+
+. User requests access through `skuba auth login` with Dex server URL, login username and password.
+. Dex receives login username and password, and do login and approve access request to the connected identity providers(connectors).
```
Comment (cclhsu) [inline adoc/admin-security-rbac.adoc]: Dex uses received login username and password to login and approve access request to the connected identity providers(connectors).

**Comment by nkoranova** [comment]
@jhsiaosuse could you please rebase the changes from master onto your branch in order to stop the Travis build from failing? 
In your skuba-auth-login branch, run: `git rebase master && git push --force-with-lease`

**Comment by jhsiaosuse** [comment]
the same, error occurs

**Pull request merged**